test/integration/publish.test.ts

```diff
@@ -1,84 +1,93 @@
+])
 
 describe('upload tarballs', async () => {
-  const cwd = process.cwd()
-  let sha: string
-  let bucket: string
-  let basePrefix: string
   beforeEach(async () => {
-    sha = await gitSha(process.cwd(), {short: true})
-    pjson.version = `${pjson.version}-${testRun}`
-    pjson.oclif.update.node.version = process.versions.node
-    bucket = pjson.oclif.update.s3.bucket
-    basePrefix = pjson.oclif.update.s3.folder
-    await deleteFolder(bucket, `${basePrefix}/versions/${pjson.version}/`)
-    await qq.writeJSON(pjsonPath, pjson)
-    const root = qq.join(__dirname, '../tmp/test/publish')
-    await qq.emptyDir(root)
-    qq.cd(root)
+    await folderCleanup()
+    await fs.writeJSON(pjsonPath, pjson)
+    await fs.emptyDir(root)
   })
   afterEach(async () => {
-    await deleteFolder(bucket, `${basePrefix}/versions/${pjson.version}/`)
-    qq.cd([__dirname, '..'])
+    await folderCleanup()
     pjson.version = originalVersion
-    await qq.writeJSON(pjsonPath, pjson)
+    await fs.writeJSON(pjsonPath, pjson)
   })
 
   skipIfWindows
-  .command(['pack:tarballs'])
-  .do(async () => {
-    sha = await gitSha(cwd, {short: true})
-  })
-  .command(['upload:tarballs'])
-  // .command(['promote', '--channel', 'stable', '-t', 'darwin-x64', '--sha', gitShaSync(process.cwd(), {short: true}), '--version', pjson.version])
-  .it('promotes valid releases', async () => {
-    const manifest = async (path: string, nodeVersion: string) => {
-      const list = await aws.s3.listObjects({Bucket: bucket, Prefix: `${basePrefix}/${path}`})
-      const manifestFile = list.Contents?.map(listObject => listObject.Key).find(f => f!.includes(target) && f!.endsWith('-buildmanifest'))
-      if (!manifestFile) {
-        throw new Error(`could not find a buildmanifest file for target ${target}`)
-      }
-
-      const manifest = await qq.readJSON(`https://${developerSalesforceCom}/${manifestFile}`)
-      const test = async (url: string, expectedSha: string, nodeVersion: string) => {
-        const xz = url.endsWith('.tar.xz')
-        const ext = xz ? '.tar.xz' : '.tar.gz'
-        await qq.download(url, `oclif${ext}`)
-        const receivedSha = await qq.hash('sha256', `oclif${ext}`)
-        expect(receivedSha).to.equal(expectedSha)
-        if (xz) {
-          await qq.x('tar xJf oclif.tar.xz')
-        } else {
-          await qq.x('tar xzf oclif.tar.gz')
-        }
-
-        const stdout = await qq.x.stdout('./oclif/bin/oclif', ['--version'])
-        expect(stdout).to.contain(`oclif/${pjson.version} ${target} node-v${nodeVersion}`)
-        await qq.rm('oclif')
-      }
-
-      await test(manifest.gz, manifest.sha256gz, nodeVersion)
-      await test(manifest.xz, manifest.sha256xz, nodeVersion)
-    }
-
+  .command(['pack:tarballs', '--parallel', '--xz'])
+  .command(['upload:tarballs', '--xz'])
+  .command(['promote', '--channel', pjson.version, '--sha', sha, '--version', pjson.version])
```

Comment:
I wouldn't want to create a new channel for every version just to keep things orderly in the s3 bucket, let's just use a `test` channel.

```suggestion
  .command(['promote', '--channel', 'test', '--sha', sha, '--version', pjson.version])
```